File: app/src/main/java/com/denchic45/kts/data/model/room/DateConverter.java
package com.denchic45.kts.data.model.room;

import androidx.room.TypeConverter;

import com.denchic45.kts.utils.DateFormatUtil;

import java.util.Date;

public class DateConverter {

    @TypeConverter
    public String toString(Date date) {
        return DateFormatUtil.convertDateToString(date, DateFormatUtil.yyy_MM_dd);
    }

    @TypeConverter
    public Date toDate(String date) {
        return DateFormatUtil.convertStringToDateUTC(date, DateFormatUtil.yyy_MM_dd);
    }

}

File: app/src/main/java/com/denchic45/kts/data/model/room/DateConverter.kt
package com.denchic45.kts.data.model.room

import androidx.room.TypeConverter
import com.denchic45.kts.utils.*
import java.util.*

class DateConverter {
    @TypeConverter
    fun toString(date: Date): String {
        return date.toString(DatePatterns.yyy_MM_dd)
    }

    @TypeConverter
    fun toDate(date: String): Date {
        return date.toDate(DatePatterns.yyy_MM_dd)
    }
}